src/main/java/xyz/nucleoid/extras/mixin/relay/PlayerManagerMixin.java

```diff
@@ -0,0 +1,32 @@
+package xyz.nucleoid.extras.mixin.relay;
+
+import net.minecraft.network.MessageType;
+import net.minecraft.server.PlayerManager;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Text;
+import org.jetbrains.annotations.Nullable;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import xyz.nucleoid.extras.event.PlayerSendChatEvent;
+
+import java.util.UUID;
+
+@Mixin(value = PlayerManager.class, priority = 2000)
+public abstract class PlayerManagerMixin {
+    @Shadow
+    @Nullable
+    public abstract ServerPlayerEntity getPlayer(UUID uuid);
+
+    @Inject(method = "broadcastChatMessage", at = @At(value = "RETURN"))
+    private void onBroadcastChatMessage(Text message, MessageType type, UUID senderUuid, CallbackInfo ci) {
+        if (type != MessageType.CHAT) return;
+        ServerPlayerEntity playerEntity = getPlayer(senderUuid);
+        if (playerEntity != null) {
+            String content = ((TranslatableTextInvoker)message).invokeGetArg(1).getString();
```

Comment:
This potentially should handle the case where there is no 1st argument. Maybe even check the 0th argument to make sure it is the player name? Not sure if that's a good idea or not.

Reply:
Yeah, that had crossed my mind. I do make sure the message is `MessageType.Chat` though so _theoretically_ it should always exist and be player name.

Reply:
I'll add a check when I get a chance